modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java

```diff
@@ -180,6 +184,66 @@ public void onCancel(ClusterNode node, GridQueryCancelRequest msg) {
         }
     }
 
+    /**
+     * Logs detailed information about a query that encountered an error during execution.
+     *
+     * @param reqId Request ID of the query.
+     * @param label Query label, if provided.
+     * @param schemaName Schema name under which the query was executed.
+     * @param queries Collection of SQL queries involved in the execution.
+     * @param params Query parameters, if any.
+     * @param error Exception that occurred during the query execution.
+     */
+    static String buildQueryLogDetails(
+            long reqId,
+            String label,
+            String schemaName,
+            Collection<GridCacheSqlQuery> queries,
+            Object[] params,
+            Throwable error,
+            UUID remoteNodeId,
+            UUID localNodeId
+    ) {
+        SensitiveDataLogging sensitivity = S.getSensitiveDataLogging();
+
+        String queriesStr = F.isEmpty(queries)
+                ? "N/A"
+                : queries.stream()
+                .map(GridCacheSqlQuery::query)
+                .collect(Collectors.joining("; "));
+
+        String paramsStr;
```

Comment:
Let's extract paramString resolution to separate utility method (i.e. `String stringifyParams(Object[] params)`). This will help us to get rid of `paramsStr` variable because use it only once in final formatting. So this method can be a pure function like:

    String stringifyParams(Object[] params) {
        if (params == null || params.length == 0)
            return "N/A";    // It's safe to tell that there were no params

        switch (S.getSensitiveDataLogging()) {
            case NONE:
                return "HIDDEN";
            case HASH:
                return String.valueOf(IgniteUtils.hash(Arrays.hashCode(params)));
            case PLAIN:
            default:
                return Arrays.toString(params);
        }    
    }